client/common/file.c

```diff
@@ -1301,6 +1301,24 @@ BOOL freerdp_client_populate_settings_from_rdp_file(rdpFile* file, rdpSettings*
 		if (!freerdp_settings_set_bool(settings, FreeRDP_SmartSizing,
 		                               (file->SmartSizing == 1) ? TRUE : FALSE))
 			return FALSE;
+		/**
+		 *  SmartSizingWidth and SmartSizingHeight:
+		 *
+		 *  Adding this option to use the DesktopHeight	and DesktopWidth as
+		 *  parameters for the SmartSizingWidth and SmartSizingHeight, as there
+		 *  are no options for that in standard RDP files.
+		 *
+		 *  Equivalent of doing /smart-sizing:WxH
+		 */
+		if (~(file->DesktopWidth) && ~(file->DesktopHeight) && file->SmartSizing == 1)
```

Comment:
this check should read `if (((~(file->DesktopWidth) && ~(file->DesktopHeight)) || ~(file->DesktopSizeId)) && (file->SmartSizing == 1))`